api/src/main/java/org/apache/iceberg/expressions/SerializationProxies.java

```diff
@@ -35,10 +35,10 @@
     /**
      * Constructor for Java serialization.
      */
-    public ConstantExpressionProxy() {
+    ConstantExpressionProxy() {
     }
 
-    public ConstantExpressionProxy(boolean trueOrFalse) {
+    ConstantExpressionProxy(boolean trueOrFalse) {
```

Comment:
seems like parameter names can have the same name as the field for the constructor. Would it make sense to keep this consistent with across constructors and method names? [e.g for the case I commented to above]